staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -1460,6 +1460,19 @@ func RunWatchSemantics(ctx context.Context, t *testing.T, store storage.Interfac
 				createdPods = append(createdPods, out)
 			}
 
+			if len(createdPods) > 0 {
+				listObject := &example.PodList{}
+				opts := storage.ListOptions{
+					Predicate:            storage.Everything,
+					Recursive:            true,
+					ResourceVersion:      createdPods[len(createdPods)-1].ResourceVersion,
+					ResourceVersionMatch: metav1.ResourceVersionMatchNotOlderThan,
+				}
+				err := store.GetList(ctx, fmt.Sprintf("/pods/%s", ns), opts, listObject)
+				require.NoError(t, err)
+				require.Equal(t, len(listObject.Items), len(createdPods))
```

Comment:
Nit: Since we're already using `require`, https://pkg.go.dev/github.com/stretchr/testify/require#Len would provide more specific failure text here.